vidarr-prometheus/src/main/java/ca/on/oicr/gsi/vidarr/prometheus/PrometheusAlertManagerConsumableResource.java

```diff
@@ -0,0 +1,167 @@
+import ca.on.oicr.gsi.cache.ReplacingRecord;
+import ca.on.oicr.gsi.cache.ValueCache;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceProvider;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import ca.on.oicr.gsi.vidarr.JsonBodyHandler;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PrometheusAlertManagerConsumableResource implements ConsumableResource {
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final ObjectReader READER = MAPPER.readerFor(new TypeReference<List<String>>() {});
+  private static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+
+  public static ConsumableResourceProvider provider() {
+    return () ->
+        Stream.of(
+            new Pair<>("prometheus-alert-manager", PrometheusAlertManagerConsumableResource.class));
+  }
+
+  private Set<String> fixedVidarrNames;
+  private String alertmanagerUrl;
+  private String environment;
+  private Set<String> labels;
+  @JsonIgnore private AlertCache cache;
+
+  public PrometheusAlertManagerConsumableResource() {}
+
+  public Set<String> getFixedVidarrNames() {
+    return fixedVidarrNames;
+  }
+
+  public String getAlertmanagerUrl() {
+    return alertmanagerUrl;
+  }
+
+  public String getEnvironment() {
+    return environment;
+  }
+
+  public Set<String> getLabels() {
+    return labels;
+  }
+
+  @Override
+  public void startup(String name) {
+    if (fixedVidarrNames.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Fixed Vidarr names is empty in Prometheus Alertmanager resource config.");
+    }
+    if (alertmanagerUrl == null || "".equals(alertmanagerUrl)) {
+      throw new IllegalArgumentException(
+          "Alertmanager URL is missing in Prometheus Alertmanager resource config.");
+    }
+    if (environment == null || "".equals(environment)) {
+      throw new IllegalArgumentException(
+          "Environment is missing in Prometheus Alertmanager resource config.");
+    }
+    if (labels.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Labels are missing in Prometheus Alertmanager resource config.");
+    }
+    cache = new AlertCache(name);
+  }
+
+  @Override
+  public Optional<Pair<String, BasicType>> inputFromSubmitter() {
+    return Optional.of(new Pair<>("required_services", BasicType.JSON));
```

Comment:
`BasicType.JSON` will let the submitter put whatever here. Shouldn't this be a `BasicType.STRING.asList()`?

Reply:
Oh I like that much better. Fixed.